Update CodeQL Schedule to scan release branches

jentfoo opened the pull request:
This allows us to switch our release branches to also use the scheduled CodeQL check so we can remove CodeQL from the PR flow for our release branches too.

While testing this PR I also tested how much speed we can gain by using go dependency caching.  That commit was retained in this PR after witnessing a 2 minute improvement (10% of the build time).

Done as part of this issue: https://github.com/gravitational/SecOps/issues/269

zmb3 reviewed (approved):
Looks good but please update the post release checklist on slab as this will have to be kept up to date when a new major is released.

jentfoo committed:
Use go dependency caching with auto build

This change speeds up the build by about ~2 minutes (10%)

jentfoo committed:
Update CodeQL Schedule to scan release branches

This allows us to switch our release branches to also use the scheduled CodeQL check so we can remove CodeQL from the PR flow for our release branches too.

jentfoo commented:
Thank you @zmb3, I added that the needed step to the Slab post.
